feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/ui/PreviewScreenComponents.kt

```diff
@@ -751,153 +846,205 @@ fun CaptureButton(
-                        val uiState = currentUiState.value
-                        when (uiState) {
-                            // stop recording after button is lifted
-                            is CaptureButtonUiState.Enabled.Recording.PressedRecording -> {
-                                onStopVideoRecording()
+                                    CaptureMode.IMAGE_ONLY -> {}
+                                }
                             }
+                        },
+                        onPress = {
+                            isPressedDown = true
+                            awaitRelease()
+                            isPressedDown = false
+                            isLongPressing = false
+                            val uiState = currentUiState.value
+                            when (uiState) {
+                                // stop recording after button is lifted
+                                is CaptureButtonUiState.Enabled.Recording.PressedRecording -> {
+                                    if (shouldBeLocked()) {
+                                        onLockVideoRecording(true)
+                                        switchPosition = 1f
+                                    } else {
+                                        onStopVideoRecording()
+                                    }
+                                }
 
-                            is CaptureButtonUiState.Enabled.Idle,
-                            CaptureButtonUiState.Unavailable -> {
-                            }
+                                is CaptureButtonUiState.Enabled.Idle,
+                                CaptureButtonUiState.Unavailable -> {
+                                }
 
-                            CaptureButtonUiState.Enabled.Recording.LockedRecording -> {}
-                        }
-                    },
-                    onTap = {
-                        val uiState = currentUiState.value
-                        when (uiState) {
-                            is CaptureButtonUiState.Enabled.Idle -> {
-                                if (!isLongPressing) {
-                                    when (uiState.captureMode) {
-                                        CaptureMode.STANDARD,
-                                        CaptureMode.IMAGE_ONLY -> onCaptureImage()
-
-                                        CaptureMode.VIDEO_ONLY -> {
-                                            onLockVideoRecording(true)
-                                            onStartVideoRecording()
+                                CaptureButtonUiState.Enabled.Recording.LockedRecording -> {}
+                            }
+                        },
+                        onTap = {
+                            val uiState = currentUiState.value
+                            when (uiState) {
+                                is CaptureButtonUiState.Enabled.Idle -> {
+                                    if (!isLongPressing) {
```

Comment:
I wonder that this already has a `onLongPress` lambda(line 889), but why here to check Long press too by variable isLongPressing?
Can't control a condition in `onLongPress`?